cardinal/ecs/storage/redis/storage.go

```diff
@@ -0,0 +1,6 @@
+package redis
+
+type EngineStorage interface {
```

Comment:
This could just be the RedisStorage

```
type RedisStorage struct {
	Namespace string
	Client    *redis.Client
	Log       zerolog.Logger
        NonceStorage
        SchemaStorage
}
```

Reply:
I don't think this will work. RedisStorage is polymorphic and we store references to RedisStorage as two different types: NonceStorage and SchemaStorage. 

RedisStorage should not contain references to itself which is sort of what's going on here. 

Unless this is what you want. This can work but Nonce and Schema are essentially pointers to itself.

Reply:
Then there's three owners of client and this makes it unnecessarily nested as we have to create two new structs just to hold the client and methods and these structs won't be reused anywhere else.

I think a better way is just have `world` hold a single:
```
type EngineStorage interface {
	NonceStorage
	SchemaStorage
}
```

And this interface just serves to hide the properties of the struct. This is more straightforward then polymorphing `RedisStorage` or creating nested structures. We can rename `EngineStorage` if needed too. What do you think?

Reply:
So i'm implementing the initial change Scott decided on. Ultimately I'm ok with this change although I prefer my suggestion over this one. Overall it will work and it's easier to follow for developers as they won't be led astray in code following by the interfaces.